python-avd/pyavd/_eos_designs/shared_utils/filtered_tenants.py

```diff
@@ -261,12 +277,56 @@ def filtered_vrfs(self: SharedUtils, tenant: dict) -> list[dict]:
             ]
 
             if vrf["svis"] or vrf["l3_interfaces"] or vrf["loopbacks"] or self.is_forced_vrf(vrf):
-                filtered_vrfs.append(vrf)
+                selected_vrfs.append(vrf)
+            else:
+                not_selected_vrfs.append(vrf)
 
             if tenant_evpn_vlan_bundle := get(tenant, "evpn_vlan_bundle"):
                 for svi in vrf["svis"]:
                     svi["evpn_vlan_bundle"] = get(svi, "evpn_vlan_bundle", default=tenant_evpn_vlan_bundle)
 
+        return selected_vrfs, not_selected_vrfs
+
+    @cached_property
+    def _build_vrfs(self: SharedUtils) -> [list[dict], list[dict]]:
+        """Return two lists of VRFs object as dicts.
+
+        Filtering based on svi tags, l3interfaces, loopbacks or self.is_forced_vrf() check.
+
+        The first one containing VRFs to be configured on the device.
+
+        The second one a list of VRFs not retained by local criteria.
+        This list is reused in case some VRFs need to be forced on the device because of the MLAG peer.
+        """
+        selected_vrfs = not_selected_vrfs = []
+
+        filter_tenants = self.filter_tenants
+        for network_services_key in self.network_services_keys:
+            tenants = get(self.hostvars, network_services_key["name"])
+            for tenant in tenants:
+                if tenant["name"] not in filter_tenants and "all" not in filter_tenants:
+                    continue
+                tenant_selected_vrfs, tenant_not_selected_vrfs = self._build_tenant_vrfs(tenant)
+                selected_vrfs.extend(tenant_selected_vrfs)
+                not_selected_vrfs.extend(tenant_not_selected_vrfs)
+
+        return selected_vrfs, not_selected_vrfs
+
+    def filtered_vrfs(self: SharedUtils, tenant: dict) -> list[dict]:
+        """
+        Return sorted and filtered vrf list from given tenant.
+
+        Keys of VRF data model will be converted to lists.
+        """
+        selected_vrfs, not_selected_vrfs = self._build_vrfs
+        filtered_vrfs = [vrf for vrf in selected_vrfs if vrf["tenant"] == tenant["name"]]
```

Comment:
Lets make sure the model we return from _build_vrfs is better suited, like a dict keyed by tenant name.